### ISSUE TITLE
GetRailwayFolder uses unintended folder

### ISSUE THREAD

**Issue by zbx1425** [leezer3/OpenBVE#505]
### Description
In the current implemention of [GetRailwayFolder](https://github.com/leezer3/OpenBVE/blob/master/source/OpenBVE/System/Loading.cs#L99), it seems that folders that have subfolders named "Railway" have priority over folders containing "Object", "Route" and "Sound" subfolders.
However, I noticed that this can led to unintended behavior under certain folder structure.
A player has told me that openBVE cannot find all his Object files. It turned out that since he did not want the files of each individual routes to mix together, he set up his directory tree like this:
```
Japan
+---Railway
|   +---Object ...
|   +---Route ...
|   \---Sound ...
+---JR UtsunomiyaLine
|   +---Object ...
|   +---Route
|       +---Utsunomiya
|           \---561M.csv
|   \---Sound ...
...
```
When he loads `Japan\JR UtsunomiyaLine\Utsunomiya\561M.csv`, since folders named "Railway" have priority, `Japan\Railway\Object` is used as the object folder, instead of the intended `Japan\JR UtsunomiyaLine\Object`. And causes the entire route to fail to load.

I think this method is not very robust. If one stores the route files outside a Railway folder, and a folder at the upstream of the tree happened to have the name of "Railway", and happened to have some content, then openBVE will be mislead.
Maybe we can try to find the railway folder by both the methods, then compare and use the directory that is "nearer" to the route file selected.

### Reproduction
The issue can be reproduced by recreating that directory structure.

**Comment by leezer3** [comment]
Changing this a great deal is likely to be problematic; It's evaluated well before the route parser starts running, and at the minute there's no way to evaluate alternate candidates if loading an object fails.
For that matter, there are plenty of routes with missing objects, and I'm not sure that re-evaluating the railway folder if we fail to load an object is likely to be a sensible idea :/

I suppose we might be able to change the Railway folder to a list of candidates, but that somewhat runs the risk of loading unintended objects in this case.

TLDR:
Get rid of the Railway folder in the directory tree & this structure will work.
May or may not be worth the effort to do something more about it, but it'll be complicated and run the risk of breaking other stuff.

**Comment by zbx1425** [comment]
I would go off-topic for a bit, the BVE5 file structure seems to eliminate the problem.
So I think problems regarding this might disappear if BVE5 route format is officially supported. But I understand there are lots of work to be done, and it is not of priority.
I really hope openBVE can enable developers write routes based on the format of BVE5 as an alternative to the current BVE4 format, but still able to utilize some openBVE-specific functions.

**Comment by zbx1425** [comment]
Maybe you can attach "wontfix" tag to issues like this.
